common/json/src/test/java/com/bazaarvoice/emodb/common/json/JsonStreamingArrayParserTest.java

```diff
@@ -67,19 +67,19 @@ public void testEOFException() {
     @Test
     @SuppressWarnings("unchecked")
     public void testMalformedChunkException() throws Exception {
-        InputSupplier<InputStream> input = ByteStreams.join(
-                ByteStreams.newInputStreamSupplier("[5,6".getBytes(Charsets.UTF_8)),
+        ByteSource input = ByteSource.concat(
+                ByteSource.wrap("[5,6".getBytes(Charsets.UTF_8)),
                 exceptionStreamSupplier(new MalformedChunkCodingException("Bad chunk header")));
-        assertThrowsEOFException(input.getInput(), Integer.class);
+        assertThrowsEOFException(input.openStream(), Integer.class);
     }
 
     @Test
     @SuppressWarnings("unchecked")
     public void testTruncatedChunkException() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.qa.bazaarvoice.com/coding_rules#rule_key=squid%3AS00112)